catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -115,6 +121,15 @@ public UpdateRequest process(UpdateRequest input)
                     getValidationProblems(validatorName, e, errors, warnings, counter);
                 }
             }
+
+            List<Serializable> tags = new ArrayList<>();
```

Comment:
you could split this out and get rid of the boolean by moving the add valid tag to right after the `validator.validate()` and the add invalid to right after the `catch (ValidationException e)`

The declaration and `addAll` would have to go right before the loop (where the boolean is declared and assigned now)